staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queueset
+
+import (
+	"time"
+
+	"k8s.io/apiserver/pkg/util/promise"
+)
+
+// request is a temporary container for "requests" with additional tracking fields
+// required for the functionality FQScheduler
+type request struct {
+	Queue *queue
+
+	// StartTime is the clock time when the request began executing
+	StartTime time.Time
+
+	// Decision gets set to the decision about what to do with this request
+	Decision promise.LockingMutable
+
+	// ArrivalTime is when the request entered this system
+	ArrivalTime time.Time
+
+	// IsWaiting indicates whether the request is presently waiting in a queue
+	IsWaiting bool
+
+	// descr1 and descr2 are not used in any logic but they appear in
+	// log messages
+	descr1, descr2 interface{}
```

Comment:
These don't have to be fields, right ?

Reply:
I am not sure what sense of "have to" you mean.  If you are suggesting that every piece of code that reads those fields has another way to get those values then I am surprised, because it did not look that way to me when I wrote this.  Look at the callers of the code that reads these field, the call stack does not necessarily include the handler handling this particular request.